### ISSUE TITLE
Ampache plugin does not work anymore

### ISSUE THREAD

**Issue by Khymick** [lotan/rhythmbox-ampache#6]
Hi there,

I am trying to use the ampache plugin on my system, but unfortunately,
it does not work. Details about the system:

Fedora 23
rhythmbox-3.3-2.fc23.x86_64
rhythmbox-ampache-0-0.5.20150518git7415a69.fc23.x86_64

It correctly connects to my Ampache server, but when I try to start
playback of any song, it simply doesn´t work. Left of the song title, a
"forbidden" symbol is showed.

If you need any logs, let me know: I will provide them.

Greets,
Michael

**Comment by lotan** [comment]
I'd recommend running rhythmbox with the debug flag.
Either
```rhythmbox -d```
for full debugging or filtered, like
```rhythmbox -D impl_open```
After you click on a song, the latter should show you the URI that the ampache-rhythmbox plugin hands over to the rhythmbox gstreamer player.
Try that URI in a web browser on the same machine and see if you can download (or play, depending on your settings) the song that way.
If that doesn't work, the problem lies within the plugin. Then try and figure out what's wrong with the URI and tell me and will do my best to fix the problem.
If it works, then the rhythmbox player settings prevent direct access to the song. It might be a proxy problem or something else to do with the network configuration.

**Comment by Khymick** [comment]
Hi there,

I started with

$ rhythmbox -d

When I try to start a song, it gives me the following output:

(10:35:25) [0x5613558c9d90] [window_focus_cb] rb-mmkeys-plugin.c:169: window got focus, re-grabbing media keys
(10:35:25) [0x5613558c9d90] [rb_shell_clipboard_entryview_changed_cb] rb-shell-clipboard.c:678: entryview changed
(10:35:25) [0x5613558c9d90] [rb_shell_clipboard_sync] rb-shell-clipboard.c:504: syncing clipboard
(10:35:26) [0x5613558c9d90] [rb_entry_view_row_activated_cb] rb-entry-view.c:2140: row activated
(10:35:26) [0x5613558c9d90] [rb_entry_view_row_activated_cb] rb-entry-view.c:2144: emitting entry activated
(10:35:26) [0x5613558c9d90] [rb_shell_player_entry_activated_cb] rb-shell-player.c:1645: got entry 0x561357409b20 activated
(10:35:26) [0x5613558c9d90] [impl_open] rb-player-gst.c:837: setting new uri to https://my.ampache.local:443/music/play/index.php?ssid=ae0dbde93c7aa78454c5b1a31766a493&type=song&oid=17236&uid=102&name=AC-DC%20-%20Have%20a%20Drink%20on%20Me.mp3
(10:35:26) [0x5613558c9d90] [impl_play] rb-player-gst.c:910: not in transition, stopping current track to start the new one
(10:35:26) [0x5613558c9d90] [start_state_change] rb-player-gst.c:406: changing state to READY
(10:35:26) [0x5613558c9d90] [start_state_change] rb-player-gst.c:410: state change succeeded synchronously
(10:35:26) [0x5613558c9d90] [state_change_finished] rb-player-gst.c:445: setting new playback URI https://my.ampache.local:443/music/play/index.php?ssid=ae0dbde93c7aa78454c5b1a31766a493&type=song&oid=17236&uid=102&name=AC-DC%20-%20Have%20a%20Drink%20on%20Me.mp3
(10:35:26) [0x5613558c9d90] [start_state_change] rb-player-gst.c:406: changing state to PLAYING
(10:35:26) [0x5613558c9d90] [rb_shell_player_set_playing_entry] rb-shell-player.c:880: Success!
(10:35:26) [0x5613558c9d90] [bus_cb] rb-player-gst.c:531: got error from stream: Aufbau einer sichern Verbindung ist gescheitert. (gstsouphttpsrc.c(1556): gst_soup_http_src_parse_status (): /GstPlayBin:playbin0/GstURIDecodeBin:uridecodebin1/GstSoupHTTPSrc:source:
Nicht akzeptables TLS-Zertifikat (6), URL: https://my.ampache.local:443/music/play/index.php?ssid=ae0dbde93c7aa78454c5b1a31766a493&type=song&oid=17236&uid=102&name=AC-DC%20-%20Have%20a%20Drink%20on%20Me.mp3, Redirect to: (NULL))
(10:35:26) [0x5613558c9d90] [track_change_done] rb-player-gst.c:359: track change failed: Aufbau einer sichern Verbindung ist gescheitert.
(10:35:26) [0x5613558c9d90] [rb_shell_player_error] rb-shell-player.c:2443: playback error while playing: Aufbau einer sichern Verbindung ist gescheitert.
(10:35:26) [0x5613558c9d90] [error_cb] rb-shell-player.c:2547: exiting error hander
(10:35:26) [0x5613558c9d90] [start_state_change] rb-player-gst.c:406: changing state to NULL
(10:35:26) [0x5613558c9d90] [start_state_change] rb-player-gst.c:410: state change succeeded synchronously
(10:35:26) [0x5613558c9d90] [rb_shell_player_handle_eos] rb-shell-player.c:498: called to simulate EOS for playing entry, but nothing is playing
(10:35:27) [0x5613558c9d90] [rb_audioscrobbler_should_handshake] rb-audioscrobbler.c:839: No username set
(10:35:27) [0x5613558c9d90] [rb_audioscrobbler_should_handshake] rb-audioscrobbler.c:839: No username set

Generelly, I can copy & paste the link in a browser and open it with eg vlc.

What may give a hint on the cause of the problem is the following line:

[bus_cb] rb-player-gst.c:531: got error from stream: Aufbau einer sichern Verbindung ist gescheitert. (gstsouphttpsrc.c(1556): gst_soup_http_src_parse_status (): /GstPlayBin:playbin0/GstURIDecodeBin:uridecodebin1/GstSoupHTTPSrc:source:
Nicht akzeptables TLS-Zertifikat (6), URL: https://my.ampache.local:443/music/play/index.php?

Translated, this means that my TLS-Certificate is not accepted - perhaps solving this may take playing songs back to live. But as I am not that deep into how rhythmbox and/or the plugin can accept certificates, it would help if you could tell me how to possibly fix this.

Greets,
Michael

**Comment by lotan** [comment]
From my point of view you have three options:
- Is your web server certificate invalid? Make it a valid certificate.
  You can have a closer look using the openssl command or simply have a look through your browser (on firefox double-click on the lock at the left hand side of the address bar).
   It may be that it's invalid due to time constraints if it worked before. Creating a new one with a new validity period should make access possible again.
- You could try to disable certificate validation on the rhythmbox side by disabling https://developer.gnome.org/libsoup/stable/SoupSession.html#SoupSession--ssl-strict, but you would have to patch and rebuild rhythmbox (CAUTION: DANGEROUS)
- If you're on the local net (as your url suggests), you could just forgo encryption and directly access http. Be aware, however, that people on the local net can monitor network traffic and extract information.

**Comment by Khymick** [comment]
Hi there,

to your solutions:

- my certs are valid ; anyhow, I refreshed them, but still the problems stay the same.
- disabling certification validation is certainly not the way it should go. As you said: Dangerous.
- I temporarily enabled http to the ampache server, now it works as it should.

I got myself behind some research and here what´s I found as solution:

- Under Ubuntu (I used 15.10):

Copy the public part of the webservers on the Ubuntu system as root at the right place:

$ sudo cp public.pem /usr/share/ca-certificates/public.crt

!!! Note the changing of the ending of the filename!!!

Reconfigure the cert-store:

$ sudo dpkg-reconfigure ca-certificates

Select your own cert in the store and go "OK"

- Fedore (I used 23):

Copy the public part of the webservers on the Fedora system as root at the right place:

$ cp public.pem /etc/pki/ca-trust/source/anchors/

Reconfigure the cert-store as root:

$ update-ca-trust
